astroquery/eso/core.py

```diff
@@ -296,248 +218,472 @@ def _login(self, *, username: str = None, store_password: bool = False,
+              record count for the specified filters.
+            - When ``query_str_only`` is ``True``, returns the query string that
+              would be issued to the TAP service given the specified arguments.
         """
-
-        url = "http://archive.eso.org/wdb/wdb/adp/phase3_main/form"
-        if open_form:
-            webbrowser.open(url)
-        elif help:
-            self._print_surveys_help(url, cache=cache)
-        else:
-            survey_form = self._request("GET", url, cache=cache)
-            query_dict = kwargs
-            query_dict["wdbo"] = "csv/download"
-            if isinstance(surveys, str):
-                surveys = surveys.split(",")
-            query_dict['collection_name'] = surveys
-            if self.ROW_LIMIT >= 0:
-                query_dict["max_rows_returned"] = int(self.ROW_LIMIT)
-            else:
-                query_dict["max_rows_returned"] = 10000
-
-            survey_response = self._activate_form(survey_form, form_index=0,
-                                                  form_id='queryform',
-                                                  inputs=query_dict, cache=cache)
-
-            content = survey_response.content
-            # First line is always garbage
-            content = content.split(b'\n', 1)[1]
-            log.debug("Response content:\n{0}".format(content))
-            if _check_response(content):
-                table = Table.read(BytesIO(content), format="ascii.csv",
-                                   comment="^#")
-                return table
-            else:
-                warnings.warn("Query returned no results", NoResultsWarning)
-
-    def query_main(self, *, column_filters={}, columns=[],
-                   open_form=False, help=False, cache=True, **kwargs):
+        _ = open_form, cache  # make explicit that we are aware these arguments are unused
+        column_filters = column_filters if column_filters else {}
+        up = _UserParams(table_name=_EsoNames.phase3_table,
+                         column_name=_EsoNames.phase3_surveys_column,
+                         allowed_values=surveys,
+                         cone_ra=cone_ra,
+                         cone_dec=cone_dec,
+                         cone_radius=cone_radius,
+                         columns=columns,
+                         column_filters=column_filters,
+                         top=top,
+                         count_only=count_only,
+                         query_str_only=query_str_only,
+                         print_help=help,
+                         authenticated=authenticated,
+                         )
+        t = self._query_on_allowed_values(user_params=up)
+        t = reorder_columns(t, DEFAULT_LEAD_COLS_PHASE3)
+        return t
+
+    @deprecated_renamed_argument(('open_form', 'cache'), (None, None),
+                                 since=['0.4.11', '0.4.11'])
+    def query_main(
```

Comment:
maybe this is a good time to rename this method to be more descriptive? Maybe `query_instruments`?

Then we can fully deprecate the single instrument query from below, as that would just be a special case for running this query